src/transport/RendezvousParameters.h

```diff
@@ -80,8 +81,8 @@ class RendezvousParameters
 
 private:
     Optional<NodeId> mLocalNodeId; ///< the local node id
-    uint32_t mSetupPINCode  = 0;   ///< the target peripheral setup PIN Code
-    uint16_t mDiscriminator = 0;   ///< the target peripheral discriminator
+    uint32_t mSetupPINCode = 0;    ///< the target peripheral setup PIN Code
+    int32_t mDiscriminator = -1;   ///< the target peripheral discriminator
```

Comment:
Why have you converted it from `uint16_t` to `int32_t` ? Is it in order to have a default value that is `!= 0` ?

If that's the case, can we rely on the fact that `mDiscriminator` in reality is a 12-bits value and so if we use a default value such a `0xFFFF` it should be invalid ?

In this case you can add some checks to `SetDiscriminator` and verify the that the value is between the authorised value and you won't need `SafeInt`.

Reply:
Sounds better, I'll do that instead.

Reply:
How's that now @vivien-apple ?

Reply:
What about adding some code into `SetDiscriminator` as well ? In this case we are sure nobody can ever set a discriminator that it invalid, and we won't have to worry on someone setting it to `UINT16_MAX - 1` :)

Reply:
I didn't add a check there because there was no error return in `SetDiscriminator()`. I thought it was redundant, I can add something there too if you'd like. 
Why is `UINT16_MAX - 1` a problem here? `HasDiscriminator()` is checking for `<= 0xFFF`.

Reply:
For now I've added a constant and filed https://github.com/project-chip/connectedhomeip/issues/3631